py3status/modules/xrandr_toggle.py

```
@@ -0,0 +1,170 @@
+    position = "left-of DVI-I-1"
+    format = "[{screen}:TV32\"]"
+}
+```
+
+@author mrt-prodz
+"""
+from subprocess import call, check_output
+from re import search
+from time import time
+
+
+class Py3status:
+    # available configuration parameters
+    cache_timeout = 10
+    format = '{source}:{status}'
+    format_disconnected = 'disconnected'
+    format_on = 'on'
+    format_off = 'off'
+    number = '1'
+    position = ''
+    source = ''
+
+    def __init__(self):
+        self.monitor = None
+        self.monitors = self._xrandr()
+
+    # return error occurs
+    def _error(self, text, color=None):
+        if color is None:
+            color = '#FF0000'
+        response = {
+            'cached_until': time() + self.cache_timeout,
+            'full_text': 'xrandr_toggle: ' + text,
+            'color': color
+        }
+        return response
+
+    # query xrandr for monitors status
+    def _xrandr(self):
+        try:
+            status = check_output(['xrandr']).split('\n')
+            monitors = []
+            # width x height + left + top
+            regex = r'\d{1,}x\d{1,}\+\d{1,}\+\d{1,}'
+            # for each line find (dis)connected monitors
+            for line in status:
+                if 'connected' in line:
+                    details = line.split(' ')
+                    if len(details) > 2:
+                        source = {}
+                        source['id'] = details[0]
+                        if details[1] == 'disconnected':
+                            source['connected'] = 0
+                            source['state'] = 'off'
+                        else:
+                            source['connected'] = 1
+                            if search(regex, details[2]):
+                                source['state'] = 'auto'
+                            else:
+                                source['state'] = 'off'
+                        monitors.append(source)
+            return monitors
+        except:
```

The thing with this is we don't know the issue.  Maybe I don't have `xrandr` installed or it had some issue.  I think the `xrandr` output is fairly standardised so I'd just let errors be thrown, that way we can fix any parsing issues.  At the moment we don't get very good feedback from `py3status` of the error.  I've got some patches to improve on this but I want #226 to get merged first.